quinn-proto/src/transport_parameters.rs

```diff
@@ -145,6 +170,9 @@ macro_rules! make_struct {
             /// of transport parameter extensions.
             /// When present, it is included during serialization but ignored during deserialization.
             pub(crate) grease_transport_parameter: Option<ReservedTransportParameter>,
+
+            /// The order in which transport parameters are serialized
+            pub(crate) write_order: Option<[u8; SUPPORTED_TRANSPORT_PARAMETERS.len()]>,
```

Comment:
Not really mandatory, I though `write_order` only make sense for instances of `TransportParameters` which are created to be then send, so received `TransportParameters` has `None` as `write_order`. I can change code to avoid `Option`, should I?

Reply:
Fair -- feels a little ugly that we're sticking it into `TransportParameters` but I see your point that it's not trivial to pass the RNG around to where we need it. Can you at least more clearly document on the field why it is `Option` and when it is expected to be `Some` vs `None`? (Keep the current comment on its own line, then an empty line, then the additional context.)